templates/class.mustache

```diff
@@ -96,11 +96,17 @@ export class {{&className}} {
      
         req = req.query(queryParameters);
 
+        if (this.requestHeadersHandler) {
```

Comment:
The move of this is to ensure this doesn't overwrite the content-type header below right?

Reply:
This is one reason, exactly. The other reason is that the `req.send` method checks the headers if the body is of type `string`.

If the headers not set at this point, `superagent` assumes that the body is form payload and transforms the body string to an object assuming it's in the format `a=b&foo=Bar`.

That's why moving the headers was also necessary as it turned out.